heat/core/signal.py

```diff
@@ -45,9 +45,11 @@ def convolve(a: DNDarray, v: DNDarray, mode: str = "full") -> DNDarray:
         Contrary to the original `numpy.convolve`, this function does not
         swap the input arrays if the second one is larger than the first one.
         This is because `a`, the signal, might be memory-distributed,
-        whereas the filter `v` is assumed to be non-distributed,
-        i.e. a copy of `v` will reside on each process.
+        whereas the filter `v` may or may not be distributed,
+        i.e. a copy of `v` may reside on each process.
```

Comment:
Now we can deal with distributed kernels, so we can improve consistency with numpy by swapping `a` and `v` if `a` is smaller than `v`, correct?

Reply:
yes, will make the required changes.